packages/next/pages/_document.tsx

```diff
@@ -344,6 +344,25 @@ export class Head extends Component<
 
     return (
       <head {...this.props}>
+        {this.context._documentProps.isDevelopment &&
+          this.context._documentProps.hasCssMode && (
+            <>
+              <style
+                data-next-hydrating
+                dangerouslySetInnerHTML={{
```

Comment:
What's the reason for using dangerouslySetInnerHTML here rather than a text node?